batch.py

```diff
@@ -357,10 +831,97 @@ def manage_concurrency_for_job(job: hb.job.Job):
             window_size=window_size,
             plink_file=plink_file,
         )
+        genotype_jobs.append(plink_job)
+
+    # the next phase will be done for each cell type
+    celltype_list = celltypes.split(' ')
+    print(celltype_list)
+    for celltype in celltype_list:
+        expression_tsv_path = dataset_path(
+            os.path.join(
+                expression_files_prefix,
+                'expression_files',
+                f'{celltype}_expression.tsv',
+            )
+        )
+
+        genes_list = extract_genes(genes_of_interest, expression_tsv_path)
+        print(f"Genes to run: {genes_list}")
+        if len(genes_list) == 0:
+            print("No genes to run!")
+            continue
+        gene_run_jobs = []
+        pv_files = []
+        for gene in genes_list:
+
+            pv_file = f'{celltype}/{gene}_pvals.csv'
+            print(pv_file)
+            if to_path(pv_file).exists():
+                print("We already ran associations for this gene!")
+                continue
+
+            print(f"Preparing inputs for: {gene}")
+            print(gene_dict[gene]['plink'])
+            if gene_dict[gene]['plink'] is None:
+                print("No plink files for this gene, exit!")
```

Comment:
This should be a logging statement, and include information about the gene in the warning.